src/OpenTelemetry/Metrics/MeterProviderBuilderExtensions.cs

```diff
@@ -107,7 +108,20 @@ public static MeterProviderBuilder AddView(this MeterProviderBuilder meterProvid
             if (metricStreamConfiguration is ExplicitBucketHistogramConfiguration histogramConfiguration)
             {
                 // Validate histogram boundaries
-                if (histogramConfiguration.Boundaries != null && !IsSortedAndDistinct(histogramConfiguration.Boundaries))
+                if (histogramConfiguration.BucketBounds == null ||
+                    !HasValidHistogramBucketBounds(histogramConfiguration.BucketBounds))
+                {
+                    throw new ArgumentException("Histogram bounds must be in ascending order with distinct values. double.NaN is not allowed.", nameof(histogramConfiguration.BucketBounds));
+                }
+
+                // If double.NegativeInfinity or double.PositiveInfinity exist in the BucketBounds, skip the value.
+                if (histogramConfiguration.BucketBounds != null)
```

Comment:
A user can provide just `double.NegativeInfinity` and `double.PositiveInfinity` as the `Boundaries` and it should be accepted. The SDK should treat that as one bucket `(-inf, +inf)`.

Reply:
Correct, will replace the `||` on line 120 with `&&`